dlt/common/libs/pydantic.py

```diff
@@ -121,14 +128,32 @@ def pydantic_to_table_schema_columns(
             # try to coerce unknown type to text
             data_type = "text"
 
-        if data_type == "complex" and skip_complex_types:
+        # This case is for a single field schema/model
+        # we need to generate snake_case field names
+        # and return flattened field schemas
+        if is_inner_type_pydantic_model and skip_complex_types:
```

Comment:
hmmm I assume that it should be `is_inner_type_pydantic_model and not skip_complex_types`? if we have `skip_complex_types` we leave the task of inferring the complex type (no matter if there's a model or not) to the normalizer.
@codingcyclist what would you expect? if we have a pydantic model defined, should we ignore `skip_complex_types` or not?

Reply:
@rudolfix I updated it this way. Will wait for @codingcyclist's feedback.

Reply:
@rudolfix you're exactly right, thank's for catching this one. It should be `is_inner_type_pydantic_model and not skip_complex_types`, exactly as I mentioned it in [my previous comment](https://github.com/dlt-hub/dlt/pull/901#discussion_r1470269082). This should also be caught by one of the two test cases that we have